### ISSUE TITLE
NotImplementedYetException when inferring property node

### ISSUE THREAD

**Issue by wryk** [rectorphp/rector#6539]
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 0.11.30 - 53697af64d37628704946ae3ebd134231b2e5a08 |

I got a `Rector\Core\Exception\NotImplementedYetException` raised by `Rector\TypeDeclaration\TypeInferer\PropertyTypeInferer\AllAssignNodePropertyTypeInferer::inferProperty` when applying `Rector\TypeDeclaration\Rector\Property\PropertyTypeDeclarationRector`.

After some debugging and nodes printed I isolated the culprit node `public $nonDefinedProperty;` (not the actual name ofc) with the codebase looking like

```php
public function bar()
{
    // ...
    $foo = $this->nonDefinedProperty->someMethod(/* ... */);
    // ...
}
```
## Minimal PHP Code Causing Issue

I tried to reproduce from scratch based on the rules previously applied before the error but the demo app only show `Fatal error: Syntax error ` so I don't know if I got the right one

https://getrector.org/demo/1ebd9e1d-67b6-64aa-b685-cfb4072d02e3

## Expected Behaviour

The error isn't helpful but I think Rector can't fix the code because it's broken in my case (dead code never running). The only non broken case allowing to use a non defined property I can think of it's when the property is assigned outside of its class.

```php
class Foo {
	
	public function baz()
	{
		return $this->nonDefinedProperty->format(\DateTimeInterface::ISO8601);
	}
}

$foo = new Foo();
$foo->nonDefinedProperty = new \DateTime();

$foo->baz();
```

But it looks pretty dumb in the first place.

**Comment by TomasVotruba** [comment]
I'll check what is wrong with the demo. Thank for reporting!

In the meantime, could you send a failing test case in a pull-request, so we have it covered in Rector?
Here is step by step tutorial how to add it: https://github.com/rectorphp/rector/blob/master/docs/how_to_add_test_for_rector_rule.md

**Comment by samsonasik** [comment]
It seems already fixed in latest 0.11.35 ref https://getrector.org/demo/1ebe0103-dc24-68c0-8b13-3fe5507cafbc
